### PR TITLE
Update to Gradle 4.2, Guava 23.1, ktlint 0.9.2, checkstyle 8.2

### PR DESCRIPTION
**Pull request by AustinShalit**

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
The `gradlew` wrapper and the `gradle.bat` file didn't change as a part of this? Double check on that please.

**Review by JLLeitschuh — COMMENTED** [review]
See my comment.

File: build.gradle.kts [hunk 1]
```diff
@@ -51,7 +51,7 @@ dependencies {
     compile(group = "edu.wpi.first.ntcore", name = "ntcore-java", version = "4.+")
     compile(group = "edu.wpi.first.wpiutil", name = "wpiutil-java", version = "3.+")
     compile(group = "org.controlsfx", name = "controlsfx", version = "8.40.14")
-    compile(group = "com.google.guava", name = "guava", version = "23.0")
+    compile(group = "com.google.guava", name = "guava", version = "23.1-jre")
```
Comment (JLLeitschuh) [inline build.gradle.kts:54]: That's a weird version specifier

Comment (AustinShalit) [reply build.gradle.kts:54]: Yea.  They changed it.

**Comment by AustinShalit** [comment]
>The gradlew wrapper and the gradle.bat file didn't change as a part of this? Double check on that please.

It is all set.  Looks like they did not change.

File: build.gradle.kts (hunk 1 above)
Comment (AustinShalit) [reply build.gradle.kts:54]: https://groups.google.com/forum/#!topic/guava-discuss/jN49PDKWzqI/discussion

**Review by JLLeitschuh — DISMISSED** [review]
These changes look good to me!

**Comment by PeterJohnson** [comment]
Do we really need to update to 4.2.1 at this point?  I'd really prefer to keep all of our repos on the same version.  Everywhere we're using 4.x right now, we're using 4.1.

**Comment by JLLeitschuh** [comment]
> I'd really prefer to keep all of our repos on the same version.

Why?

**Review by JLLeitschuh — DISMISSED** [review]
This looks fine to me.

**Pull request closed without merging**